internal/internal.go

```diff
@@ -9,27 +9,34 @@ import (
 	"crypto/rand"
 	"encoding/json"
 	"fmt"
-	"github.com/open-policy-agent/opa/topdown"
 	"io"
+	"io/ioutil"
 	"net"
 	"net/url"
 	"strconv"
 	"strings"
 	"sync"
 	"time"
 
+	"github.com/golang/protobuf/protoc-gen-go/descriptor"
+	"github.com/jhump/protoreflect/desc"
+	"github.com/jhump/protoreflect/dynamic"
+	"github.com/open-policy-agent/opa/topdown"
+
 	ctx "golang.org/x/net/context"
 
-	ext_core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
-	ext_authz "github.com/envoyproxy/go-control-plane/envoy/service/auth/v2"
-	ext_type "github.com/envoyproxy/go-control-plane/envoy/type"
+	ext_core "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	ext_authz "github.com/envoyproxy/go-control-plane/envoy/service/auth/v3"
+	ext_type "github.com/envoyproxy/go-control-plane/envoy/type/v3"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 	"google.golang.org/genproto/googleapis/rpc/code"
 	rpc_status "google.golang.org/genproto/googleapis/rpc/status"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/reflection"
 
+	proto "github.com/golang/protobuf/proto"
```

Comment:
[nit] it's already called proto, so we should be able to do:
```suggestion
	"github.com/golang/protobuf/proto"
```